src/pybind/mgr/snap_schedule/module.py

```diff
@@ -0,0 +1,178 @@
+"""
+Copyright (C) 2019 SUSE
+
+LGPL2.1.  See file COPYING.
+"""
+import errno
+import json
+import sqlite3
+from .fs.schedule_client import SnapSchedClient
+from mgr_module import MgrModule, CLIReadCommand, CLIWriteCommand
+from mgr_util import CephfsConnectionException
+from threading import Event
+
+
+class Module(MgrModule):
+
+    def __init__(self, *args, **kwargs):
+        super(Module, self).__init__(*args, **kwargs)
+        self._initialized = Event()
+        self.client = SnapSchedClient(self)
+
+    def resolve_subvolume_path(self, fs, subvol, path):
+        if not subvol:
+            return path
+
+        rc, subvol_path, err = self.remote('fs', 'subvolume', 'getpath',
+                                           fs, subvol)
+        if rc != 0:
+            # TODO custom exception?
+            raise Exception(f'Could not resolve {path} in {fs}, {subvol}')
+        return subvol_path + path
+
+    @property
+    def default_fs(self):
+        fs_map = self.get('fs_map')
+        if fs_map['filesystems']:
+            return fs_map['filesystems'][0]['mdsmap']['fs_name']
+        else:
+            self.log.error('No filesystem instance could be found.')
+            raise CephfsConnectionException(
+                -errno.ENOENT, "no filesystem found")
+
+    def serve(self):
+        self._initialized.set()
+
+    def handle_command(self, inbuf, cmd):
+        self._initialized.wait()
+        return -errno.EINVAL, "", "Unknown command"
+
+    @CLIReadCommand('fs snap-schedule status',
+                    'name=path,type=CephString,req=false '
+                    'name=subvol,type=CephString,req=false '
+                    'name=fs,type=CephString,req=false',
+                    'List current snapshot schedules')
+    def snap_schedule_get(self, path='/', subvol=None, fs=None):
+        use_fs = fs if fs else self.default_fs
+        try:
+            ret_scheds = self.client.get_snap_schedules(use_fs, path)
+        except CephfsConnectionException as e:
+            return e.to_tuple()
+        return 0, '\n===\n'.join([ret_sched.report() for ret_sched in ret_scheds]), ''
```

Comment:
ditto as per list: any reason why this is not a json out?

Reply:
ditto `--format json`